content/updates/understanding-the-terraform-check-block-feature.md

```diff
@@ -0,0 +1,210 @@
+2. **Out-of-band change detection:** an unanticipated change made outside Terraform can cause surprising failures, especially if it can't be detected via a built-in drift detection mechanism (for instance, when another root module manages a resource). Checks can come in handy to ensure deviations such as the expiration of a TLS certificate or authentication tokens won't create unforeseen interruptions.
+3. **Enforcing naming and tagging conventions:** maintaining a consistent set of tagging and naming conventions within your organization can be crucial. Use the `check` block feature to enforce these standards across your infrastructure, fostering uniformity and orderliness.
+4. **Upholding security best practices:** checks can help ensure adherence to security best practices within your infrastructure. For example, you can use it to verify that specific ports are closed, IAM configurations are properly set, and encryption is enabled, contributing to a more secure and robust environment.
+
+## Do I need this if I'm already testing my Terraform code?
+
+While similar in their goals of ensuring correct and efficient infrastructure operation, infrastructure validation, and testing differ slightly in their scopes and methods.
+
+_Infrastructure validation_ generally refers to checking the correctness of infrastructure code against specified standards or requirements before it's deployed. For instance, native Terraform commands like `validate` or `plan` or tools like [TFLint](https://github.com/terraform-linters/tflint) mainly focus on the correctness of code and check the syntax and internal consistency, ensuring it's well-structured and free from apparent errors before deployment.
+
+_Infrastructure testing_, on the other hand, usually takes place after the infrastructure has been provisioned and validates that it's functioning as expected. This can include unit tests, integration tests, functional tests, and acceptance tests, among others. Testing tools like [Kitchen-Terraform](https://newcontext-oss.github.io/kitchen-terraform/) and [Terratest](https://terratest.gruntwork.io/) are commonly used for these kinds of tests. They can confirm that resources are adequately created and connected, security groups allow/deny correct traffic, etc.
+
+Both infrastructure validation and testing are essential for a healthy development process. Here is where the `check` feature comes into play and might feed two birds with one crumb.
+
+First, checks can work great when the actual resource provisioning isn't the primary focus of the assertion. Given the necessity of configuring sandbox environments (whether ephemeral or persistent, hosted in the cloud, or run locally using mocking tools) and writing tests in Ruby or Go, native HCL `check` blocks can be seen as a practical alternative.
```

Comment:
This 2nd sentence was confusing and I think we can drop it? Let me know if you disagree. 
```suggestion
First, checks can work great when the actual resource provisioning isn't the primary focus of the assertion.
```

Reply:
Yeah, I'm not opposed to that. Initially, I wanted to emphasize that a user needs to think about provisioning a testing environment, while checks are run against a "real" one. But this text was rewritten a lot, and this thought probably has lost its value and clarity.